oauthproxy.go

```diff
@@ -50,6 +50,7 @@ var SignatureHeaders = []string{
 	"Date",
 	"Authorization",
 	"X-Forwarded-User",
+	"X-Forwarded-Groups",
```

Comment:
This is needed -- but it will also be a breaking change to existing users that use the GAP-Signature functionality for upstream signature verification.  We just need to add a note to the changelog.

How many users actually user this... I don't know.  Hence I made this Issue: https://github.com/oauth2-proxy/oauth2-proxy/issues/648